Doc/library/pathlib.rst

```diff
@@ -150,23 +150,49 @@ we also call *flavours*:
    to ``PurePosixPath('bar')``, which is wrong if ``foo`` is a symbolic link
    to another directory)
 
+   The optional *template* argument may provide another path object. It is
+   supplied whenever a new path object is created from an existing one, such
+   as in :attr:`parent` or :meth:`relative_to`. Subclasses may use this to
+   pass information between path objects. For example::
+
+      from pathlib import PurePosixPath
+
+      class MyPath(PurePosixPath):
+          def __init__(self, *pathsegments, template=None, session_id=None):
+              super().__init__(*pathsegments, template=template)
+              if template:
+                  self.session_id = template.session_id
+              else:
+                  self.session_id = session_id
```

Comment:
I've written this out in full for clarity's sake, but it could be:

```suggestion
              self.session_id = template.session_id if template else session_id
```

Reply:
I personally prefer the full version because the modern python coverage tools cannot handle ternary expressions.